### PR TITLE
HPCC-16656 Debug Regression

### PR DESCRIPTION
**Pull request by richardkchapman**
Debugger broken in 6.0.x for hthor queries.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
Could target 6.0.8 if we do another rc before gold (or 6.0.10 if there is one)

@GordonSmith Please review

**Comment by GordonSmith** [comment]
Tested and this didn't fix the issue.

File: ecl/eclagent/eclagent.cpp [hunk 1]
```diff
@@ -4200,15 +4200,17 @@ class CHThorDebugGraphManager : extends CBaseDebugGraphManager
...
+    IRoxieProbe *createProbe(IInputBase *in, IActivityBase *sourceAct, IActivityBase *targetAct, unsigned sourceIdx, unsigned targetIdx, unsigned iteration) override
     {
         CriticalBlock b(crit);
         unsigned channel = debugContext->queryChannel();
         unsigned sourceId = sourceAct->queryId();
         unsigned targetId = targetAct->queryId();
         DebugActivityRecord *sourceActRecord = noteActivity(sourceAct, iteration, channel, debugContext->querySequence());
         DebugActivityRecord *targetActRecord = noteActivity(targetAct, iteration, channel, debugContext->querySequence());
-        DebugProbe *probe = new DebugProbe(dynamic_cast<IHThorInput*>(in), stream, sourceId, sourceIdx, sourceActRecord, targetId, targetIdx, targetActRecord, iteration, channel, debugContext);
+        IHThorInput *hin = dynamic_cast<IHThorInput*>(in);
```
Comment (GordonSmith) [inline ecl/eclagent/eclagent.cpp]: Is this legit when debugging on roxie?

**Review by GordonSmith — APPROVED** [review]
One question, looks ok otherwise.

File: ecl/eclagent/eclagent.cpp (hunk 1 above)
Comment (richardkchapman) [inline ecl/eclagent/eclagent.cpp]: This code is only in eclagent

**Pull request merged**